### ISSUE TITLE
Project: Onboarding Flow

### ISSUE THREAD

**Issue by mchowning** [Automattic/pocket-casts-android#524]
## Description

Add a new onboarding flow to help users get up to speed more quickly.

### Initial Setup
- [X] Put feature behind feature flag during development ([#525](https://github.com/Automattic/pocket-casts-android/pull/525))
- [X] Create basic navigation framework with empty screens ([#525](https://github.com/Automattic/pocket-casts-android/pull/525))
- [X] Only show onboarding to users a single time ([#603](https://github.com/Automattic/pocket-casts-android/pull/603))

### Initial choose whether to Log In or Sign Up Screen
- [X] Implement designs ([#555](https://github.com/Automattic/pocket-casts-android/pull/555), [#617](https://github.com/Automattic/pocket-casts-android/pull/617))
    - [X] Make podcasts in image dynamic based on the user's subscribed podcasts ([#665](https://github.com/Automattic/pocket-casts-android/pull/665))
- [X] Add functionality (buttons should link to appropriate screens) ([#525](https://github.com/Automattic/pocket-casts-android/pull/525))
- [X] Analytics ([#555](https://github.com/Automattic/pocket-casts-android/pull/555), [#564](https://github.com/Automattic/pocket-casts-android/pull/564), [#642](https://github.com/Automattic/pocket-casts-android/pull/642))

### Log in screen
- [X] Implement design  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Enable log in  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Handle errors  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Forgot password screen ([#560](https://github.com/Automattic/pocket-casts-android/pull/560))
- [X] Analytics  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555), [#560](https://github.com/Automattic/pocket-casts-android/pull/560), [#646](https://github.com/Automattic/pocket-casts-android/pull/646))
- [X] [v2] Migrate old log in screen to use new screen written in compose ([#694](https://github.com/Automattic/pocket-casts-android/pull/694))

### Create Account screen
- [X] Implement design  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Disable submission button until input is valid  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Handle errors  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555))
- [X] Analytics  ([#555](https://github.com/Automattic/pocket-casts-android/pull/555), [#646](https://github.com/Automattic/pocket-casts-android/pull/646))
- [X] [v2] Migrate old create account screen to use new screen written in compose ([#694](https://github.com/Automattic/pocket-casts-android/pull/694))

### Upgrade Flow
- [X] Screen: initial
    - [X] Implement designs ([#584](https://github.com/Automattic/pocket-casts-android/pull/584))
    - [X] Add functionality ([#584](https://github.com/Automattic/pocket-casts-android/pull/584))
    - [X] Analytics ([#584](https://github.com/Automattic/pocket-casts-android/pull/584))
- [X] Screen: subscription selection ([#594](https://github.com/Automattic/pocket-casts-android/pull/594))
    - [X] Implement designs ([#594](https://github.com/Automattic/pocket-casts-android/pull/594))
    - [X] Add functionality ([#594](https://github.com/Automattic/pocket-casts-android/pull/594))
    - [X] Analytics ([#648](https://github.com/Automattic/pocket-casts-android/pull/648))
    - [X] Upgrade bottom sheet with design changes ([#662](https://github.com/Automattic/pocket-casts-android/pull/662))
- [X] Make screen background expand into insets at top of screen ([comment](https://github.com/Automattic/pocket-casts-android/issues/524#issuecomment-1344439271), [#656](https://github.com/Automattic/pocket-casts-android/pull/656)).

### Welcome Screen
- [X] Screen: Welcome for free users ([#601](https://github.com/Automattic/pocket-casts-android/pull/601)]
- [X] Screen: Welcome for plus users ([#601](https://github.com/Automattic/pocket-casts-android/pull/601)]
- [X] Discover new podcasts card ([#601](https://github.com/Automattic/pocket-casts-android/pull/601)]
- [X] Analytics ([#648](https://github.com/Automattic/pocket-casts-android/pull/648))
- [ ] [v2] add confetti 🎉 (normal confetti: [#655](https://github.com/Automattic/pocket-casts-android/pull/655), plus confetti: TBD)

### Import ([#634](https://github.com/Automattic/pocket-casts-android/pull/634))
- [X] Implement design for initial import screen
- [X] Create instruction screen for Castbox
- [X] Create instruction screen for OPML import
- [X] Add import card to onboarding welcome screen
- [X] Analytics ([#648](https://github.com/Automattic/pocket-casts-android/pull/648))

### Recommendations screen
- [X] Implement design ([#604](https://github.com/Automattic/pocket-casts-android/pull/604), [#640](https://github.com/Automattic/pocket-casts-android/pull/640))
- [X] Subscribing to highlighted podcasts functionality ([#640](https://github.com/Automattic/pocket-casts-android/pull/640))
- [X] Subscribing via search functionality ([#604](https://github.com/Automattic/pocket-casts-android/pull/604))
- [X] Analytics ([#646](https://github.com/Automattic/pocket-casts-android/pull/646))
- [X] [v2] Add additional categories (Arts, Business, etc.) and Featured Podcasts ([comment](https://github.com/Automattic/pocket-casts-android/issues/524#issuecomment-1347960181), [Categories PR](https://github.com/Automattic/pocket-casts-android/pull/675), [Featured PR](https://github.com/Automattic/pocket-casts-android/pull/677))

### [v2] Upgrade box on main app's create account screen
- [ ] Initial view on create account screen
- [ ] Bottom modal (shown after tapping on initial view)

### [v2] Other entry points for Create Account Flow
- [X] Link to new upgrade flow from Profile tab's create account/log in box ([#694](https://github.com/Automattic/pocket-casts-android/pull/694))
- [X] Link to new upgrade flow from any other plus upgrade triggers in the app ([#694](https://github.com/Automattic/pocket-casts-android/pull/694))

### Wrap up
- [x] Remove the feature flag once development of v1 is complete ([#645](https://github.com/Automattic/pocket-casts-android/pull/645))

### Nice to Haves
- [X] Address issue where after switching orientations an extra back swipe is needed to exit the app ([PR comment](https://github.com/Automattic/pocket-casts-android/pull/603#pullrequestreview-1191350711)) ([#645](https://github.com/Automattic/pocket-casts-android/pull/645))

**Comment by mchowning** [comment]
The screens aren't expanding into the insets on the phone. This is probably an issue on all the screens, but it's noticeable on the plus upgrade screen.

<img width="350px" src="https://user-images.githubusercontent.com/4656348/206734747-fd24e627-1a3d-412b-9c1e-42e6f74d7aa9.jpg" />

**Comment by adamzelinski** [comment]
Hey, @mchowning just leaving some feedback here regarding the latest build you sent me. Feel free to move these somewhere else if this isn't the best spot.

Firstly, it's looking great thank you! I only noticed a couple of things which I'll list here. I know some you may still be working on.

**Login**
<img width="582" alt="Screen Shot 2022-12-13 at 6 15 38 pm" src="https://user-images.githubusercontent.com/74938219/207253637-6f0be4e5-b87f-4dcb-8e7d-43c7d6c1ed83.png">

- Now that we're hiding the social buttons, I think we could slightly adjust the padding of the Headings so that it's horizontally centered between the artwork and buttons.
- The artwork looks a bit off but I know you're working on this
- I also noticed some screens aren't expanding into the insets on the phone or the header and footer are different colors to the UI. Not sure if this is just my device but if you wouldn't mind taking a look that would be awesome.

**Upgrade Modal**
<img width="649" alt="Screen Shot 2022-12-13 at 6 17 12 pm" src="https://user-images.githubusercontent.com/74938219/207254905-54b8b201-f911-4147-ab5d-f073d97a00e9.png">

- (Bug?) When arriving at the Upgrade screen I didn't get the button to upgrade.

**Import**
<img width="682" alt="Screen Shot 2022-12-13 at 6 17 24 pm" src="https://user-images.githubusercontent.com/74938219/207255311-9dee3e34-f9b4-4525-8fd5-88631049afc8.png">
<img width="713" alt="Screen Shot 2022-12-13 at 6 17 29 pm" src="https://user-images.githubusercontent.com/74938219/207255369-b753e224-98a1-401d-97db-f86870d92aca.png">
- Can we please remove the 'Welcome Back' Text from the header here?
- Just double checking that the button to open an app for import will be shown if the app is installed on your device?
- Some of the secondary and body copy is looking a little small (might just be my device) can we please check body is set to 16

**Welcome Screen**
<img width="463" alt="Screen Shot 2022-12-13 at 6 17 47 pm" src="https://user-images.githubusercontent.com/74938219/207260367-e7d45018-564a-49b6-bde0-5f7a5ef1ddd9.png">
- (Bug) When I tapped on 'Find my next podcast' it took me back to the login screen instead closing and exiting the onboarding and displaying the discover view.

**Recommendations**
<img width="1014" alt="Screen Shot 2022-12-13 at 7 08 25 pm" src="https://user-images.githubusercontent.com/74938219/207260747-8ab125d7-f7c5-4fdf-a256-e73bd714c742.png">
- Can we please use the Find Podcasts heading when the listener scrolls this view and the heading gets hidden.
- (Bug) When I searched for a podcast and tapped to subscribe here and closed the search the button still read 'Not Now' instead of continue.
- (Not a blocker) but the idea for the recommendations view was to show two rows from the Trending, Most Popular, and then our different categories. Then when you tape view more it would populate another 2 rows and so on.
<img width="514" alt="Screen Shot 2022-12-13 at 7 56 22 pm" src="https://user-images.githubusercontent.com/74938219/207272289-6d5af1e2-1fe5-462f-be91-d26c83dc8f0e.png">

**Comment by mchowning** [comment]
Thanks for the great feedback @adamzelinski !

> Login
>
> Now that we're hiding the social buttons, I think we could slightly adjust the padding of the Headings so that it's vertically centered between the artwork and buttons.

Updated in https://github.com/Automattic/pocket-casts-android/pull/656.

> The artwork looks a bit off but I know you're working on this

Ashita has [a PR](https://github.com/Automattic/pocket-casts-android/pull/660) up to address this.

> I also noticed some screens aren't expanding into the insets on the phone or the header and footer are different colors to the UI. Not sure if this is just my device but if you wouldn't mind taking a look that would be awesome.

This has been addressed in https://github.com/Automattic/pocket-casts-android/pull/656

> Upgrade Modal
>
> (Bug?) When arriving at the Upgrade screen I didn't get the button to upgrade.

This was because I gave you a debug build to test and it couldn't retrieve the available subscriptions from Google. A release build won't have this issue.

We have talked about updating the subscription modal with [the new designs](p1669080905462829/1668645351.529999-slack-C04AV00GV39), and I'm working on that.

> Import
>
> Can we please remove the 'Welcome Back' Text from the header here?

Ashita has taken care of this in [this PR](https://github.com/Automattic/pocket-casts-android/pull/655).

> Just double checking that the button to open an app for import will be shown if the app is installed on your device?

That's right, if Castbox is installed the import screen will have a button to open Castbox.

> Some of the secondary and body copy is looking a little small (might just be my device) can we please check body is set to 16

Yes, they are both P40, which is 16.

> Welcome Screen
>
> (Bug) When I tapped on 'Find my next podcast' it took me back to the login screen instead closing and exiting the onboarding and displaying the discover view.

I've only seen this issue on an emulator, let me know if you saw it on a physical device. Either way, I made a change to fix the issue in [this PR](https://github.com/Automattic/pocket-casts-android/pull/645) just to cover us in case the issue does occur on some physical devices.

> Recommendations
>
> Can we please use the Find Podcasts heading when the listener scrolls this view and the heading gets hidden.

I'm not sure exactly what you mean here. Do you mean to add the Find Podcasts heading from the search screen here too, but only after the user has started scrolling (or scrolled a certain amount), or something else?

> (Bug) When I searched for a podcast and tapped to subscribe here and closed the search the button still read 'Not Now' instead of continue.

I've updated this so that the button says "Continue" if the user is subscribed to any podcast in [this PR](https://github.com/Automattic/pocket-casts-android/pull/661). Previously, it only said continue if the user was subscribed to one of the podcasts displayed on the recommendations screen.

> (Not a blocker) but the idea for the recommendations view was to show two rows from the Trending, Most Popular, and then our different categories. Then when you tape view more it would populate another 2 rows and so on.

Thanks for clarifying. I've noted this as a "v2" improvement in the issue description above. This probably won't be included when the onboarding flow initially rolls out in 7.29.

**Comment by adamzelinski** [comment]
Awesome, thanks for this @mchowning 

> I've only seen this issue on an emulator, let me know if you saw it on a physical device.

Yes, it was on a physical device I saw this. I was using a Google Pixel

>I'm not sure exactly what you mean here. Do you mean to add the Find Podcasts heading from the search screen here too, but only after the user has started scrolling (or scrolled a certain amount)

Yeah, that's what I was thinking. This one is not a blocker at all though. Only if there's time :)

**Comment by mchowning** [comment]
👋  @adamzelinski !

>  ...the idea for the recommendations view was to show two rows from the Trending, Most Popular, and then our different categories

I'm pulling the trending podcasts from our discover feed. What other parts/categories from the discover feed do we want to include? Looking at the current feed, I assume we'd probably want things like "Top 100 of 2022"  and "Popular in [Region]", but what about sections like "Bon Appetit", "Past Tense", and the "Featured" section? Are those what you mean by "our different categories"? 

Is there anything else from our current Discover feed that you would want included?

**Comment by EmmelineKB** [comment]
@mchowning I think it would work best to pull from the actual "Categories" sections (i.e. those you see under "Browse By Category") as opposed to using Discover lists, as some of the themes can get pretty specific. This would be my personal preference.

Including "Featured" would be good, though!

I would suggest that we perhaps leave out the "Popular in" section, as the podcasts in there are not always so relevant due to how that area is populated (it is by lifetime popularity). Unless we could move forward with updating how we populate that area to freshen those lists up like we've talked about before @geekygecko?

cc: @adamzelinski

**Comment by adamzelinski** [comment]
Hey @mchowning Yes, the idea was to pull Trending, Most Popular and then the individual categories under the "Browse by Category" Section. Happy to include Featured too and also happy to leave most popular out for now as per @EmmelineKB suggestion until it can be rethought
